src/main/java/oracle/r2dbc/impl/OracleReadableMetadataImpl.java

```diff
@@ -352,10 +352,10 @@ else if (type == R2dbcType.VARBINARY) {
 
       return new OracleColumnMetadataImpl(
         type, name, nullability,
-        // The getPrecision and getScale methods return 0 for types where
+        // The getPrecision and getScale methods return 0 or -1 for types where
         // precision and scale are not applicable.
-        precision == 0 ? null : precision,
-        scale == 0 ? null : scale);
+        precision < 1 ? null : precision,
```

Comment:
A scale or precision of 0 has a specific meaning for numeric types such as NUMBER so I don't think that returning null is correct. Also, as a side note, the scale -127 for a NUMBER is a magic value that indicates that it's not set.

Reply:
The ResultSetMetaData.getPrecision(int) method is specified to return 0 only for datatypes where precision is not applicable.
https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/ResultSetMetaData.html#getPrecision(int)

I don't think Oracle JDBC should ever return a precision of 0 for a NUMBER column because precision is applicable for this type, and it is not possible to declare a NUMBER column with a precision of 0:
```sql
SQL> create table tt (value NUMBER(0));
create table tt (value NUMBER(0))
                               *
ERROR at line 1:
ORA-01727: numeric precision specifier is out of range (1 to 38)
```

For R2DBC's column metadata, we want getPrecision() to return null when precision is not applicable for a datatype. If JDBC tells us the precision is 0, this should translate to R2DBC returning a precision of null.
The same is also true for scale. JDBC is specified to return 0 if scale is not applicable, and R2DBC is specified to return null if scale is not applicable.
https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSetMetaData.html#getScale-int-

I think I should revert this change, such that it only checks for 0, and not for a negative value. If JDBC is returning a negative value, R2DBC should just expose that value, and not replace it with null. 
I added this change for JSON, but we can handle JSON in the same way as BLOB and CLOB, where precision is not applicable.

Reply:
You can have a scale of 0 for a NUMBER but if I read your code correctly you would return null instead.
https://docs.oracle.com/cd/B28359_01/server.111/b28318/datatype.htm#CNCPT313

Reply:
I see what you mean now: getScale() should return 0 for NUMBER.

Reply:
Latest commit has a test and new code to support NUMBER with a scale of 0.

Reply:
Your latest changes look fine. Thanks!